core/src/test/java/org/apache/calcite/test/ExceptionMessageTest.java

```diff
@@ -141,4 +167,29 @@ private void runQuery(String sql) throws SQLException {
           containsString("Object 'nonexistentTable' not found"));
     }
   }
+
+  @Test void testValidRelNodeQuery() throws SQLException {
+    final RelNode relNode = builder
+        .scan("test", "entries")
+        .project(builder.field("name"))
+        .build();
+    runQuery(relNode);
+  }
+
+  @Test void testRelNodeQueryException() throws SQLException {
+    try {
+      final RelNode relNode = builder
+          .scan("test", "entries")
+          .project(builder.call(SqlStdOperatorTable.ABS, builder.field("name")))
+          .build();
+      runQuery(relNode);
+      fail("Query badEntries should result in an exception");
+    } catch (RuntimeException e) {
+      assertThat(e.getMessage(),
+          equalTo("java.sql.SQLException: Error while preparing statement [\n"
+              + "LogicalProject($f0=[ABS($1)])\n"
```

Comment:
Hmm, the plan may be helpful but it is not a statement ?

Reply:
Statement in Calcite may be  include three types: Sql. RelNode .  Queryable.  This method Called prepareStatement_. My be we listen to others opinion.

Reply:
Statement in Calcite may be  include three types: Sql. RelNode .  Queryable.  This method Called prepareStatement_. My be we listen to others opinion.

Reply:
Actually, I'm +1 for Danny, maybe change "statement" to "plan" is better. But the code before is also "Error while preparing statement"

Reply:
Ok. I will change this to plan. Thanks for all of your advice.